src/scratchOrgCreate.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// third
+import { Duration } from '@salesforce/kit';
+import { SuccessResult } from 'jsforce';
+
+// Local
+import { Org } from './org';
+import { Logger } from './logger';
+import { AuthInfo } from './authInfo';
+import { Messages } from './messages';
+import { SfdxError } from './sfdxError';
+import { Connection } from './connection';
+import { SfdxProject } from './sfdxProject';
+import { Lifecycle } from './lifecycleEvents';
+import { ConfigAggregator } from './config/configAggregator';
+import {
+  authorizeScratchOrg,
+  requestScratchOrgCreation,
+  pollForScratchOrgInfo,
+  deploySettingsAndResolveUrl,
+} from './scratchOrgInfoApi';
+import SettingsGenerator from './scratchOrgSettingsGenerator';
+import { generateScratchOrgInfo, getScratchOrgInfoPayload } from './scratchOrgInfoGenerator';
+
+Messages.importMessagesDirectory(__dirname);
+const messages: Messages = Messages.loadMessages('@salesforce/core', 'scratchOrgCreate');
+
+const DEFAULT_STREAM_TIMEOUT_MINUTES = 6;
+
+export interface ScratchOrgCreateResult {
+  username?: string;
+  authInfo?: AuthInfo;
+  warnings: string[];
+}
+export interface ScratchOrgCreateOptions {
+  hubOrg: Org;
```

Comment:
I think you should jsdoc this interface.  

Examples--what does retry do?  
What happens if you don't specify an apiVersion?
Should I pass in a `definitionjson` or a `definitionfile` or both ?  What happens if I do both?
what's an orgConfig?  I wouldn't know from the type itself (Record<string, unknown>) what to put in there.  

Look at this through the lens of someone who wants to create scratch orgs from a node process outside of our commands, who doesn't know our flags.  How would they figure out what to send?